From cb69968d5d3b462ac9b7c6c1d71da32417bb301c Mon Sep 17 00:00:00 2001
From: Mitali Vagadiya <vmnagajibhai@habana.ai>
Date: Thu, 22 May 2025 12:41:23 +0300
Subject: [PATCH] [JSW-705] jgs: module: cleanup jgs firmware code

This patch removes unnecessary or additional comments,
renames fs related code to jgs and performs complete
code cleanup.

Change-Id: Ia1ecc869983210cd23e7fbb176436619f6ec26d0
---

diff --git a/arch/jgs/asic.c b/arch/jgs/asic.c
index a65fd6a..4c88eaf 100644
--- a/arch/jgs/asic.c
+++ b/arch/jgs/asic.c
@@ -1,4 +1,4 @@
 // SPDX-License-Identifier:     GPL-2.0+
 /*
- * Copyright (C) 2024 HabanaLabs Ltd.
+ * Copyright (C) 2025 Intel Ltd.
  */
diff --git a/arch/jgs/common.c b/arch/jgs/common.c
index 64b8f08..d5fe246 100644
--- a/arch/jgs/common.c
+++ b/arch/jgs/common.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier:     GPL-2.0+
 /*
- * Copyright (C) 2024 HabanaLabs Ltd.
+ * Copyright (C) 2025 Intel Ltd.
  */
 
 #include <parc_global_conf_regs.h>
@@ -47,7 +47,7 @@
 
 #ifdef __ZEPHYR__
 #include <preboot.h>
-LOG_MODULE_REGISTER(fs1_emb_common);
+LOG_MODULE_REGISTER(jgs_emb_common);
 #else
 #include <baremetal.h>
 #endif
@@ -113,7 +113,25 @@
 	return 0;
 }
 
+/* REMOVE ONCE CI PASS */
 u32 fs1_get_arc_addr_from_64bit(u64 addr)
+{
+	/* ARC.MNG DCCM and ICCM regions */
+	if (addr >= IO0_MNG_ICCM_FULL_BASE && addr < IO0_MNG_ICCM_FULL_BASE +
+		(u64)IO_0_MNG_ARC_ICCM_SIZE)
+		return (u32)((addr & ADDR_MASK(IO_MNG_ICCM)) + LOCAL_ICCM_BASE);
+
+	if (addr >= IO0_MNG_DCCM_FULL_BASE && addr < IO0_MNG_DCCM_FULL_BASE +
+		(u64)IO_0_MNG_ARC_DCCM_SIZE)
+		return (u32)((addr & ADDR_MASK(IO_MNG_DCCM)) + LOCAL_DCCM_BASE);
+
+	/* TODO Other regions better be added as well (GT, HBW, DBG) */
+
+	/* CFG region (15) */
+	return (addr & 0xffffffff);
+}
+
+u32 jgs_get_arc_addr_from_64bit(u64 addr)
 {
 	/* ARC.MNG DCCM and ICCM regions */
 	if (addr >= IO0_MNG_ICCM_FULL_BASE && addr < IO0_MNG_ICCM_FULL_BASE +
@@ -229,7 +247,7 @@
 	return 0;
 }
 
-u32 fs1_get_arc_reg_offset(u64 addr)
+u32 jgs_get_arc_reg_offset(u64 addr)
 {
 	u32 reg_offset = (u32)addr | CFG_ARC_REGION_BASE;
 
@@ -246,7 +264,7 @@
 #define ARC_RST_ADDR_SHIFT	10
 #define ARC_RUN_HALT_TIMEOUT_US	100
 
-static void fs1_arc_halt_cpu(u32 cpu_base, int type)
+static void jgs_arc_halt_cpu(u32 cpu_base, int type)
 {
 	u64 start;
 	bool ack;
@@ -282,7 +300,7 @@
 
 int arch_stop_and_reset_cpu(uint64_t base, int type, uint64_t start)
 {
-	u32 cpu_base = fs1_get_arc_reg_offset(base);
+	u32 cpu_base = jgs_get_arc_reg_offset(base);
 	bool halted = false;
 	u32 rst_addr = (u32)start >> ARC_RST_ADDR_SHIFT;
 
@@ -295,7 +313,7 @@
 	}
 	/* if arc is not halted halt the arc */
 	if (!halted)
-		fs1_arc_halt_cpu(cpu_base, type);
+		jgs_arc_halt_cpu(cpu_base, type);
 
 	/* reset, set reset vector, and release reset */
 	if (type == CPU_TYPE_PSOC) {
@@ -316,9 +334,9 @@
 #define ARC_RST_ADDR_SHIFT	10
 #define ARC_RUN_HALT_TIMEOUT_US	100
 
-void fs1_run_arc_cpu(uint64_t base, int type)
+void jgs_run_arc_cpu(uint64_t base, int type)
 {
-	u32 cpu_base = fs1_get_arc_reg_offset(base);
+	u32 cpu_base = jgs_get_arc_reg_offset(base);
 	u64 start;
 	bool ack;
 
@@ -405,7 +423,7 @@
 		}
 
 		EMB_LOG_HEX_INF("Run Non-ARC.MNG image for cpu: ", i);
-		fs1_run_arc_cpu(cpu_arr[i].cpubase, cpu_arr[i].cputype);
+		jgs_run_arc_cpu(cpu_arr[i].cpubase, cpu_arr[i].cputype);
 	}
 
 	if (local_cpu_jump) {
diff --git a/arch/jgs/onix.c b/arch/jgs/onix.c
index a65fd6a..4c88eaf 100644
--- a/arch/jgs/onix.c
+++ b/arch/jgs/onix.c
@@ -1,4 +1,4 @@
 // SPDX-License-Identifier:     GPL-2.0+
 /*
- * Copyright (C) 2024 HabanaLabs Ltd.
+ * Copyright (C) 2025 Intel Ltd.
  */
diff --git a/arch/jgs/palad.c b/arch/jgs/palad.c
index 056e238..6d510f0 100644
--- a/arch/jgs/palad.c
+++ b/arch/jgs/palad.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier:     GPL-2.0+
 /*
- * Copyright (C) 2024 HabanaLabs Ltd.
+ * Copyright (C) 2025 Intel Ltd.
  */
 
 #include <parc_global_conf_regs.h>
@@ -36,7 +36,7 @@
 
 #ifdef __ZEPHYR__
 #include <preboot.h>
-LOG_MODULE_REGISTER(fs1_emb_common_palad);
+LOG_MODULE_REGISTER(jgs_emb_common_palad);
 #else
 #include <baremetal.h>
 #endif
diff --git a/include/lib_common.h b/include/lib_common.h
index 3f1070e..c79df29 100644
--- a/include/lib_common.h
+++ b/include/lib_common.h
@@ -12,9 +12,11 @@
 
 u64 get_64bit_addr_from_32bit_addr(uintptr_t base);
 u32 fs1_get_arc_addr_from_64bit(u64 addr);
+u32 jgs_get_arc_addr_from_64bit(u64 addr);
 int hw_mem_copy(uint64_t src, uint64_t dst, uint32_t size, bool is_hbw, bool const_data);
 int hw_mem_init(uint64_t dst, uint32_t size, bool is_hbw);
 uint32_t fs1_get_arc_reg_offset(uint64_t addr);
+uint32_t jgs_get_arc_reg_offset(uint64_t addr);
 int arch_stop_and_reset_cpu(uint64_t base, int type, uint64_t start);
 int arch_mem_init(uint64_t base, uint32_t size);
 u64 get_mgmt_cpu_base(void);
